### ISSUE TITLE
`Overflow` in `tf.compat.v1.nn.ctc_loss` and `tf.compat.v1.nn.ctc_loss_v2` when there are too large values in `labels`.

### ISSUE THREAD

**Issue by xl1043237213** [tensorflow/tensorflow#65864]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

binary

### TensorFlow version

tf 2.15

### Custom code

No

### OS platform and distribution

Linux Ubuntu 20.04

### Python version

3.10

### Current behavior?

`Overflow` in `tf.compat.v1.nn.ctc_loss` and `tf.compat.v1.nn.ctc_loss_v2` when the value of `labels` is too large. See the colab link below for details.

### Standalone code to reproduce the issue

```shell
https://colab.research.google.com/drive/1zKblem8QIsfEA973gnXrY6NDLch6ysoa?usp=sharing
```

### Relevant log output

```shell
Non-OK-status: InitDims(dim_sizes) status: INVALID_ARGUMENT: Encountered overflow when multiplying 2 with 9143586203607084488, result: -159571666495382640
```

**Comment by SuryanarayanaY** [comment]
Hi @xl1043237213 ,

I have replicated the reported behaviour and attached screenshot below. The overflow warning is expected but it follows a crash which might needs fix.

<img width="1485" alt="Screenshot 2024-04-17 at 11 38 46 AM" src="https://github.com/tensorflow/tensorflow/assets/116063290/a7ec7fa2-780d-4d37-8f33-739bd678fb97">

**Comment by kiransair** [comment]
Hi @xl1043237213, Could you please try to use this `tf.keras.ops.ctc_loss` which is also a similar ops for calculating the ctc loss. 
https://www.tensorflow.org/api_docs/python/tf/keras/ops/ctc_loss
Thank You.
